libusb/libusbi.h

```diff
@@ -164,7 +164,7 @@ static inline void *usbi_reallocf(void *ptr, size_t size)
 #ifndef TIMESPEC_TO_TIMEVAL
 #define TIMESPEC_TO_TIMEVAL(tv, ts)                                     \
         do {                                                            \
-                (tv)->tv_sec = (ts)->tv_sec;                            \
+                (tv)->tv_sec = (long) (ts)->tv_sec;                            \
```

Comment:
We have to process overflow case. Throw exception if cast result wrong would be good enough.
if((long) ts->tv_sec != ts->tv_sec) throw <bad_cast>;